test/ecto/parameterized_type_test.exs

```diff
@@ -0,0 +1,143 @@
+defmodule Ecto.ParameterizedTypeTest do
+  use ExUnit.Case, async: true
+
+
+  defmodule MyParameterizedType do
+    use Ecto.ParameterizedType
+
+    @params %{some_param: :some_param_value}
+
+    def params, do: @params
+
+    def init([some_opt: :some_opt_value, field: :my_type, schema: _]), do: @params
+    def type(@params), do: :custom
+    def load(_, _, @params), do: {:ok, :load}
+    def dump( _, _, @params),  do: {:ok, :dump}
+    def cast( _, @params),  do: {:ok, :cast}
+    def equal?(true, _, @params), do: true
+    def equal?(_, _, @params), do: false
+    def embed_as(_, @params), do: :dump
```

Comment:
Since `Ecto.Enum` has embed_as `:dump`, let's make this one `:self` so we can test different scenarios.

Reply:
Done. I also added tests for the error paths (I originally added it for :maybe, but that isn't working yet).